LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift

```diff
@@ -0,0 +1,317 @@
+        ]
+
+        if supportedModelSettings.fiaspModelEnabled {
+            options.append(.exponentialPreset(.fiasp))
+        }
+
+        if supportedModelSettings.walshModelEnabled {
+            options.append(.walsh(WalshInsulinModel(actionDuration: viewModel.walshActionDuration)))
+        }
+
+        return options
+    }
+
+    private var selectedInsulinModelValues: [GlucoseValue] {
+        oneUnitBolusEffectPrediction(using: viewModel.insulinModelSettings.model)
+    }
+
+    private var unselectedInsulinModelValues: [[GlucoseValue]] {
+        selectableInsulinModelSettings
+            .filter { $0 != viewModel.insulinModelSettings }
+            .map { oneUnitBolusEffectPrediction(using: $0.model) }
+    }
+
+    private func oneUnitBolusEffectPrediction(using model: InsulinModel) -> [GlucoseValue] {
+        let bolus = DoseEntry(type: .bolus, startDate: chartManager.startDate, value: 1, unit: .units)
+        let startingGlucoseSample = HKQuantitySample(type: HKQuantityType.quantityType(forIdentifier: .bloodGlucose)!, quantity: startingGlucoseQuantity, start: chartManager.startDate, end: chartManager.startDate)
+        let effects = [bolus].glucoseEffects(insulinModel: model, insulinSensitivity: viewModel.insulinSensitivitySchedule)
+        return LoopMath.predictGlucose(startingAt: startingGlucoseSample, effects: effects)
+    }
+
+    private var startingGlucoseQuantity: HKQuantity {
+        let startingGlucoseValue = viewModel.insulinSensitivitySchedule.quantity(at: chartManager.startDate).doubleValue(for: glucoseUnit) + glucoseUnit.glucoseExampleTargetValue
+        return HKQuantity(unit: glucoseUnit, doubleValue: startingGlucoseValue)
+    }
+
+    private var endingGlucoseQuantity: HKQuantity {
+        HKQuantity(unit: glucoseUnit, doubleValue: glucoseUnit.glucoseExampleTargetValue)
+    }
+
+    private func isSelected(_ settings: InsulinModelSettings) -> Binding<Bool> {
+        Binding(
+            get: { self.viewModel.insulinModelSettings == settings },
+            set: { isSelected in
+                if isSelected {
+                    withAnimation {
+                        self.viewModel.insulinModelSettings = settings
+                    }
+                }
+            }
+        )
+    }
+
+    private var isWalshModelSelected: Binding<Bool> {
+        Binding(
+            get: { self.viewModel.insulinModelSettings.model is WalshInsulinModel },
+            set: { isSelected in
+                if isSelected {
+                    withAnimation {
+                        self.viewModel.insulinModelSettings = .walsh(WalshInsulinModel(actionDuration: self.viewModel.walshActionDuration))
+                    }
+                }
+            }
+        )
+    }
+
+    var dismissButton: some View {
+        Button(action: dismiss) {
+            Text("Close", comment: "Button text to close a modal")
+        }
+    }
+}
+
+fileprivate struct InsulinModelSettingsRow: View {
```

Comment:
and then these fileprivate structs ended up being removed in favor of the extracted component